clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -587,8 +603,8 @@ public static Errors forCode(short code) {
         if (error != null) {
             return error;
         } else {
-            log.warn("Unexpected error code: {}.", code);
-            return UNKNOWN;
+            log.warn("Unknown error code: {}.", code);
+            return UNKNOWN_RETRIABLE;
```

Comment:
Hmm, I thought the plan is for uncaught exceptions still be UNKNOWN (i.e., not retriable), but if the server throws a new exception that's retriable, the server will send UNKNOWN_RETRIABLE to old clients.